server/api/utils/background_tasks/internal.py

```diff
@@ -20,112 +20,32 @@
-        timeout: int = None,  # in seconds
-        name: str = None,
-        *args,
-        **kwargs,
-    ) -> mlrun.common.schemas.BackgroundTask:
-        name = name or str(uuid.uuid4())
-        logger.debug(
-            "Creating background task",
-            name=name,
-            project=project,
-            function=function.__name__,
-        )
-        server.api.utils.singletons.db.get_db().store_background_task(
-            db_session,
-            name,
-            project,
-            mlrun.common.schemas.BackgroundTaskState.running,
-            timeout,
-        )
-        background_tasks.add_task(
-            self.background_task_wrapper,
-            db_session,
-            project,
-            name,
-            function,
-            *args,
-            **kwargs,
-        )
-        return self.get_background_task(db_session, name, project)
-
-    def get_background_task(
-        self,
-        db_session: sqlalchemy.orm.Session,
-        name: str,
-        project: str,
-    ) -> mlrun.common.schemas.BackgroundTask:
-        return server.api.utils.singletons.db.get_db().get_background_task(
-            db_session, name, project
-        )
-
-    async def background_task_wrapper(
-        self,
-        db_session: sqlalchemy.orm.Session,
-        project: str,
-        name: str,
-        function,
-        *args,
-        **kwargs,
-    ):
-        try:
-            if asyncio.iscoroutinefunction(function):
-                await function(*args, **kwargs)
-            else:
-                await fastapi.concurrency.run_in_threadpool(function, *args, **kwargs)
-        except Exception as exc:
-            err_str = mlrun.errors.err_to_str(exc)
-            logger.warning(
-                "Background task execution failed",
-                function_name=function.__name__,
-                exc=err_str,
-                tb=traceback.format_exc(),
-            )
-            server.api.utils.singletons.db.get_db().store_background_task(
-                db_session,
-                name,
-                project=project,
-                state=mlrun.common.schemas.BackgroundTaskState.failed,
-                error=err_str,
-            )
-        else:
-            server.api.utils.singletons.db.get_db().store_background_task(
-                db_session,
-                name,
-                project=project,
-                state=mlrun.common.schemas.BackgroundTaskState.succeeded,
-            )
-
-
 class InternalBackgroundTasksHandler(metaclass=mlrun.utils.singleton.Singleton):
     def __init__(self):
         self._internal_background_tasks: typing.Dict[
             str, mlrun.common.schemas.BackgroundTask
         ] = {}
 
+        # contains a lock for each background task kind, with the following format:
+        # {kind: [active_name, previous_name]}
+        self._background_tasks_kind_locks: typing.Dict[
+            str, typing.Tuple[typing.Optional[str], typing.Optional[str]]
```

Comment:
id suggest to `kind` it as a const of kinds so that it would be easier to search usages of such

Reply:
But then we can't use templates to create custom kind strings. Enforcing the kind being in a subset of constants will limit the functionality of the feature.

Reply:
perhaps u can const the template, the idea is to provide a way to find usages of background kinds